src/main/java/redis/clients/jedis/JedisPool.java

```diff
@@ -333,21 +332,14 @@ public Jedis getResource() {
   }
 
   @Override
-  protected void returnBrokenResource(final Jedis resource) {
-    if (resource != null) {
-      returnBrokenResourceObject(resource);
-    }
-  }
-
-  @Override
-  protected void returnResource(final Jedis resource) {
+  public void returnResource(final Jedis resource) {
     if (resource != null) {
       try {
         resource.resetState();
         returnResourceObject(resource);
       } catch (Exception e) {
         returnBrokenResource(resource);
-        throw new JedisException("Resource is returned to the pool as broken", e);
+        log.debug("Resource is returned to the pool as broken", e);
```

Comment:
I think we should use `log.warn` or `log.error`, `log.debug` is generally not used in a production environment, but we need to record errors.